src/main/java/com/limechain/sync/JustificationVerifier.java

```diff
@@ -26,14 +30,23 @@
 @Log
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
```

Comment:
```suggestion
@UtilityClass
```

And then we can remove the static from flag from the methods.